cpp/src/arrow/testing/util.cc

```diff
@@ -84,8 +84,12 @@ std::string random_string(int64_t n, uint32_t seed) {
 
 void random_decimals(int64_t n, uint32_t seed, int32_t precision, uint8_t* out) {
   pcg32_fast gen(seed);
-  std::uniform_int_distribution<uint32_t> d(0, std::numeric_limits<uint8_t>::max());
-  const int32_t required_bytes = DecimalType::DecimalSize(precision);
+  // Generate smaller values than requested to avoid generating invalid decimals for the
```

Comment:
`RandomArrayGenerator` has a more precise random generation routine, should we simply switch to it?